### ISSUE TITLE
Support for `storage` directive

### ISSUE THREAD

**Issue by iromeo** [JetBrains-Research/snakecharm#554]
See https://snakemake.readthedocs.io/en/stable/snakefiles/storage.html#storage-support

Example:
```
storage fs:
    provider="fs"


rule all:
    input:
        storage("fs/test2.txt")


checkpoint foo:
    output:
        storage("fs/test.txt")
    shell:
        "echo 'test' > {output}"


rule bar:
    input:
        storage("fs/test.txt")
    output:
        storage("fs/test2.txt")
    shell:
        "cp {input} {output}"
```

`storage` is `snakemake.storage.StorageRegistry` objected, injected in `workflow.py`:
```
_globals["storage"] = self._storage_registry
```

**Comment by iromeo** [comment]
Fixed as part of #550

**Comment by iromeo** [comment]
<img width="425" alt="image" src="https://github.com/user-attachments/assets/1147d185-e9d5-4443-895b-67bbd9c02bf9">
